### ISSUE TITLE
X-OAUTH2 authentication mechanism

### ISSUE THREAD

**Issue by mremond** [FluuxIO/go-xmpp#116]
It would allow to use OAUTH bearer tokens to authenticate on ejabberd.

**Comment by mremond** [comment]
I will be making a breaking change to the Config struct, to be able to pass credentials that are not passwords.

I am planning to change config from:

```
      config := xmpp.Config{
		Address:      "localhost:5222",
		Jid:          "test@localhost",
		Password:     "test",
	}
```

to:

```
      config := xmpp.Config{
		Address:      "localhost:5222",
		Jid:          "test@localhost",
		Credential:     xmpp.Password("test"),
	}
```

This will allow us to introduce a new credential: `xmpp.OAuthToken("token")`

**Comment by mremond** [comment]
Implemented by #117
